### PR TITLE
`Development`: Add exam signature upload test

### PR DESCRIPTION
**Pull request by nityanandaz**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context
**Previously**, the Exam Checker App couldn't load the signature after it was submitted. The Test checks that the behavior stays fixed.

Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-09-20 at 01 29 16

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Exam

1. Log in to the Exam Checker App
2. Let _student1_ sign their information
3. Verify that the app can load/display the signature

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2
#### Exam Mode Test
_Planned on Monday_
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

**Comment by Strohgelaender** [comment]
@julian-christl will this cause issues with #7038?

**Comment by julian-christl** [comment]
I mean I already fixed this in #7038 as you mentioned in the meeting @nityanandaz so I'm not sure why you would want to do another PR that would only create merge conflicts ^^

**Review by krusche — APPROVED** [review]
Code looks good to me 👍

**Comment by nityanandaz** [comment]
@julian-christl we want to be sure that any fix is ready for the exam next week

**Comment by julian-christl** [comment]
Fair @nityanandaz But I sent @bassner my PR already, it's maintainer-approved and Patrick approved it so I suspect is only a matter of when he intends to hit the button.

**Review by Strohgelaender — DISMISSED** [review]
This is a test-only PR now after merging develop. The new test looks good to me.

File: src/test/java/de/tum/in/www1/artemis/FileIntegrationTest.java
Comment (laadvo) [inline src/test/java/de/tum/in/www1/artemis/FileIntegrationTest.java:106]: Could you maybe rename the variable `postResponse`? As it is now, I think it is not very clear what it actually is.

File: src/test/java/de/tum/in/www1/artemis/FileIntegrationTest.java
Comment (laadvo) [inline src/test/java/de/tum/in/www1/artemis/FileIntegrationTest.java:108]: Same for `getResponse`.

**Review by laadvo — COMMENTED** [review]
I have some suggestions on how to make the test a bit more clearer. If you decide to implement them, also consider not using `var` for all variables.
But otherwise looks good.

**Review by Strohgelaender — APPROVED** [review]

**Review by julian-christl — APPROVED** [review]

**Pull request merged**
